### PR TITLE
feat: DeliveryHog extra logging and filtering

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

More in progress work

## Changes

* Added some extra logging to the hog function stuff
* Pre-filter parsed messages based on whether the team has functions or not
* Gathers logs from the function and adds them to log entries table

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: plugin-server/src/cdp/hog-executor.ts [hunk 1]
```diff
@@ -285,15 +313,15 @@ export class HogExecutor {
...
+            // await this.executeAsyncResponse({
+            //     ...invocation,
+            //     hogFunctionId: hogFunction.id,
+            //     vmState: execResult.state!,
+            //     response: {
+            //         status: fetchResponse.status,
+            //         body: await fetchResponse.text(),
+            //     },
+            // })
         }
```
Comment (mariusandra) [inline plugin-server/src/cdp/hog-executor.ts]: No more multihop?

File: posthog/api/log_entries.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    if search:
+        clickhouse_where_parts.append("message ILIKE %(search)s")
+        clickhouse_kwargs["search"] = f"%{search}%"
+    if len(level) > 0:
+        clickhouse_where_parts.append("upper(level) in %(levels)s")
+        clickhouse_kwargs["levels"] = level
+
+    clickhouse_query = f"""
+        SELECT log_source_id, instance_id, timestamp, upper(level) as level, message FROM log_entries
+        WHERE {' AND '.join(clickhouse_where_parts)} ORDER BY timestamp DESC {f'LIMIT {limit}' if limit else ''}
```
Comment (mariusandra) [inline posthog/api/log_entries.py:79]: This without a limit (or an Optional limit) feels dangerous

File: plugin-server/src/cdp/hog-executor.ts (hunk 1 above)
Comment (mariusandra) [reply plugin-server/src/cdp/hog-executor.ts]: lol, that's what happens when you request a review too soon

File: posthog/api/log_entries.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/api/log_entries.py:79]: Good point. Will update.

**Review by mariusandra — APPROVED** [review]

**Pull request merged**